tf_agents/agents/behavioral_cloning/behavioral_cloning_agent.py

```diff
@@ -286,6 +286,12 @@ def _setup_as_continuous(self, time_step_spec, action_spec, loss_fn):
     collect_policy = actor_policy.ActorPolicy(
         time_step_spec, action_spec, actor_network=self._cloning_network)
     policy = greedy_policy.GreedyPolicy(collect_policy)
+
+    # Ensure local MSE doesn't collapse batch dimension
+    self._loss_mse_fn = tf.losses.mse
+    if tf.nest.flatten(action_spec)[0].shape.rank == 0:
```

Comment:
self._loss_mse_fn = tf.nest.map_structure(lambda s: tf.math.squared_difference if s.shape.rank==0 else tf.losses.mse, action_spec

(keep lines w/in 80 char col limit)